panml/core/llm/huggingface.py

```diff
@@ -39,9 +40,11 @@ def __init__(self, model: str, input_block_size: int, padding_length: int, token
             self.tokenizer = AutoTokenizer.from_pretrained(self.model_hf.config.tokenizer_class.lower().replace('tokenizer', ''), mirror='https://huggingface.co')
         else:
             self.tokenizer = AutoTokenizer.from_pretrained(self.model_name, mirror='https://huggingface.co')
-        
-        if self.tokenizer.pad_token is None:
-            self.tokenizer.pad_token = self.model_hf.config.eos_token_id
+
+        # Set model on GPU if available and specified
+        if 'gpu' in model_args:
+            self.device = 'cuda' if torch.cuda.is_available() and model_args['gpu'] else 'cpu'
```

Comment:
do we need to issue warning that one of these conditions arent met?

Reply:
Was thinking the default is "cpu" as that is industry standard. Are you thinking of like a print statement to say what type is used?

Reply:
i meant if 'gpu' is the user input, however the conditions in line 46 is not met, should we warn the user that we will proceed with cpu.

Reply:
otherwise the user will still think that we ran the code with gpu

Reply:
ok yeah good point, I'll do the print statement to show what the model is set to (gpu or cpu)

Reply:
i thought it's simply`.to(self.device)`?

Reply:
I thought it was to.('cuda') as well, then searched around and seems like its torch.device